src/Policy/LimitPerIp.php

```diff
@@ -10,16 +10,14 @@ final class LimitPerIp implements LimitPolicyInterface
 {
     public function fingerprint(ServerRequestInterface $request): string
     {
-        return sha1(mb_strtolower($request->getMethod() . '-' . $request
-                ->getUri()
-                ->getPath() . '-' . $this->getIp($request)));
+        return sha1(strtolower($request->getMethod() . $request->getUri()->getPath() . $this->getIp($request)));
     }
 
     private function getIp(ServerRequestInterface $request): string
     {
         /** @psalm-var array{REMOTE_ADDR?: string} $server */
         $server = $request->getServerParams();
 
-        return trim($server['REMOTE_ADDR'] ?? '', '[]');
```

Comment:
This is can be valid case. @samdark @kafkiansky can you explain why need trim here?

Reply:
It doesn't make sense to trim the `[]` chars because of `REMOTE_ADDR` presents only IP in usual form: 4 bytes separated with a dot and 8 bytes separated with a colon for IPv6.